hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/HadoopConfigurations.java

```diff
@@ -49,8 +48,7 @@ public static org.apache.hadoop.conf.Configuration getParquetConf(
    */
   public static org.apache.hadoop.conf.Configuration getHadoopConf(Configuration conf) {
     org.apache.hadoop.conf.Configuration hadoopConf = FlinkClientUtil.getHadoopConf();
-    Map<String, String> options = FlinkOptions.getPropertiesWithPrefix(conf.toMap(), HADOOP_PREFIX);
-    options.forEach(hadoopConf::set);
+    conf.toMap().forEach(hadoopConf::set);
```

Comment:
This could be a breaking change.

Reply:
Yes, the current approach is not to filter. If filtered by prefix, enumerate all such as s3a.I can't think of a better way now.

Reply:
How many s3a options you wanna pass around, what's the problem there if you prefix all those custom hadoop options with the `hadoop.` prefix.

Reply:
Does these work for you?

```java
hadoop.fs.s3a.access.key
hadoop.fs.s3a.secret.key,
hadoop.fs.s3a.endpoint,
hadoop.fs.s3a.impl.disable.cache,
hadoop.fs.s3a.impl,
hadoop.fs.s3a.connection.maximum
```

Reply:
The aws package does not recognize parameters such as the hadoop prefix, so the connection fails.

Reply:
I mean to configure them through Flink options, the Flink would trim all these prefixes for them.

Reply:
Sorry for my late reply, do you mean such filtering, 
```
    // Approach 4: Flink configuration
        // add all configuration key with prefix 'flink.hadoop.' in flink conf to hadoop conf
        for (String key : flinkConfiguration.keySet()) {
            for (String prefix : FLINK_CONFIG_PREFIXES) {
```
but hudi does not have flink.hadoop prefix cutting, even if it does, it will be before hadoop prefix filtering.Or do we write it differently, for example, we use flink.handoop to filter and crop the prefix ourselves.

Reply:
If we write down a option `hadoop.k1` in Flink SQL, Flink would filter out this option and trim it into `k1`, the `k1` option is used to construct the hadoop configuration, does that work for you?

Reply:
Useless, because we will call HadoopConfigurations. getHadoopConf (conf) filter out k1. It is closer to the underlying file system interface than the Flink option

Reply:
Am i missing something:

```java
  public static org.apache.hadoop.conf.Configuration getHadoopConf(Configuration conf) {
    org.apache.hadoop.conf.Configuration hadoopConf = FlinkClientUtil.getHadoopConf();
    Map<String, String> options = FlinkOptions.getPropertiesWithPrefix(conf.toMap(), HADOOP_PREFIX);
    options.forEach(hadoopConf::set);
    return hadoopConf;
  }
```

The `HadoopConfigurations. getHadoopConf` already handle the prefixed options.

Reply:
Oh, I'm sorry, I don't see substring in this method.This is usefully,The previous test was stuck in hivecatalog, where there was no cropping.thanks very much.